sdk/core/azure-core/azure/core/tracing/opentelemetry.py

```diff
@@ -133,13 +132,34 @@ def start_as_current_span(
         :paramtype attributes: Optional[Attributes]
         :keyword links: Links to add to the span.
         :paramtype links: Optional[Sequence[Link]]
+        :keyword end_on_exit: Whether to end the span when exiting the context manager. Defaults to True.
```

Comment:
since this will be on by default, is there some behavior change here thats different? iow should it opt-in ?

Reply:
Behavior will be the same by default since we were passing `end_on_exit=True` previously. Actually, `end_on_exit` is `True` by default in OTel-land as well, so we are adopting their default convention of ending spans upon exiting their contexts. This behavior would be more familiar to OTel users.

For most use cases, the span ending on exit is convenient since most operations are encompassed inside the span context.